modules/safer-cluster/main.tf

```diff
@@ -0,0 +1,166 @@
+  // that the necessary security features are enabled.
+  kubernetes_version = "latest"
+
+  // Nodes are created with a default version. The nodepool enables
+  // auto_upgrade so that the node versions can be kept up to date with
+  // the master upgrades.
+  //
+  // https://cloud.google.com/kubernetes-engine/versioning-and-upgrades
+  node_version = ""
+
+  master_authorized_networks_config = var.master_authorized_networks_config
+
+  subnetwork        = var.subnetwork
+  ip_range_pods     = var.ip_range_pods
+  ip_range_services = var.ip_range_services
+
+  horizontal_pod_autoscaling = var.horizontal_pod_autoscaling
+  http_load_balancing        = var.http_load_balancing
+
+  // Disable the dashboard. It creates risk by running as a very sensitive user.
+  kubernetes_dashboard = false
+
+  // We suggest the use coarse network policies to enforce restrictions in the
+  // communication between pods.
+  //
+  // NOTE: Enabling network policy is not sufficient to enforce restrictions.
+  // NetworkPolicies need to be configured in every namespace. The network
+  // policies should be under the control of a cental cluster management team,
+  // rather than individual teams.
+  network_policy          = true
+  network_policy_provider = "CALICO"
+
+  maintenance_start_time = var.maintenance_start_time
+
+  initial_node_count = var.initial_node_count
+
+  // We suggest removing the default node pull, as it cannot be modified without
+  // destroying the cluster.
+  remove_default_node_pool = true
+
+  disable_legacy_metadata_endpoints = true
+
+  node_pools        = var.node_pools
+  node_pools_labels = var.node_pools_labels
+
+  // TODO(mmontan): check whether we need to restrict these
+  // settings.
+  node_pools_metadata = var.node_pools_metadata
+  node_pools_taints   = var.node_pools_taints
+  node_pools_tags     = var.node_pools_tags
+
+  // TODO(mmontan): we generally considered applying
+  // just the cloud-platofrm scope and use Cloud IAM
+  // If we have Workload Identity, are there advantages
+  // in restricting scopes even more?
+  node_pools_oauth_scopes = var.node_pools_oauth_scopes
+
+  stub_domains         = var.stub_domains
+  upstream_nameservers = var.upstream_nameservers
+
+  // We should use IP Alias.
+  configure_ip_masq = false
+
+  logging_service    = var.logging_service
+  monitoring_service = var.monitoring_service
+
+  // We never use the default service account for the cluster. The default
+  // project/editor permissions can create problems if nodes were to be ever
+  // compromised.
+
+  // We either:
+  // - Create a dedicated service account with minimal permissions to run nodes.
+  //   All applications shuold run with an identity defined via Workload Identity anyway.
+  // - Use a service account passed as a parameter to the module, in case the user
+  //   wants to maintain control of their service accounts.
+  create_service_account = length(var.compute_engine_service_account) > 0 ? false : true
```

Comment:
Instead of using `length` here, prefer comparing it with its default value. Such as:

```
var.compute_engine_service_account == "" ? false : true
```